Embrace Unity iOS Interface/EmbraceManager.swift

```diff
@@ -0,0 +1,494 @@
+                            timestamp: Date(), // Should we let users input their own exception timestamp?
+                            attributes: attributes,
+                            stackTraceBehavior: .notIncluded)
+    }
+    
+    static func logUnhandledException(name: String, message: String, stacktrace: String) {
+        var attributes = [
+            "exception.stacktrace": stacktrace,
+            "emb.exception_handling": "unhandled",
+            "emb.type": "sys.exception",
+            "exception.message": message,
+            "exception.type": name
+        ]
+        
+        Embrace.client?.log("Unity exception",
+                            severity: .error,
+                            timestamp: Date(), // Should we let users input their own exception timestamp? That would create an API mismatch among the Hosted SDKs. Let's not.
+                            attributes: attributes,
+                            stackTraceBehavior: .notIncluded)
+    }
+    
+    static func logPushNotification(title: String, body: String, subtitle: String, badge: Int, category: String) -> Bool {
+        
+        // Borrowed from Embrace Flutter SDK
+        let pushData: [AnyHashable: Any?] = [
+                        "aps": [
+                            "alert" : [
+                                "title" : title,
+                                "subtitle" : subtitle,
+                                "body" : body
+                            ],
+                            "badge" : badge as Any,
+                            "category" : category as Any
+                        ]
+                    ]
+        do {
+            try Embrace.client?.add(event: .push(userInfo: pushData as [AnyHashable: Any]))
+            return true
+        } catch let error {
+            // We should probably log this error
+        }
+        
+        return false
+    }
+    
+    private static func transferKVPs(dest: inout [String:String], src: NSDictionary) {
+        for (key, value) in src {
+            if let key = key as? String, let value = value as? String {
+                dest.updateValue(value, forKey: key)
+            }
+        }
+    }
+    
+    private static func convertNSDictToSwiftDict<T>(nsDict: NSDictionary, converter: (String) -> T) -> [String: T] {
+        var swiftDict = [String: T]()
+        
+        for (key, value) in nsDict {
+            if let key = key as? String, let value = value as? String {
+                swiftDict.updateValue(converter(value), forKey: key)
+            } else {
+                // Should the whole operation fail or should we just log the issue?
+                // We probably should not let the whole operation fail. We can skip the value
+            }
+        }
+        
+        return swiftDict
+    }
+    
+    private static func createNetworkSpanName(url: String, httpMethod: String) -> String {
+        var name = "emb-" + httpMethod.uppercased()
```

Comment:
Is the `emb-` something we add? should a captured request look like
```
emb-GET http://localhost:1234/a-random/path
```

Reply:
This stylistically matches RN at this time: https://github.com/embrace-io/embrace-react-native-sdk/blob/62f98e3bd5d2c1de90b944cb2ae69ff7cb8ae62a/packages/core/ios/RNEmbrace/EmbraceManager.swift#L564

However, I think matching iOS here is probably preferable.

Reply:
We should probably note to come back to this.